x-pack/plugins/apm/server/types.ts

```diff
@@ -143,6 +143,7 @@ const requiredDependencies = [
   'infra',
   'observability',
   'ruleRegistry',
+  'fleet',
```

Comment:
if this is a required dependency now, we should move it from the `optionalPlugins` to `requiredPlugins` in `x-pack/plugins/apm/kibana.json`, and we can also get rid of checks for the existence of the plugin in places like here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/server/routes/settings/agent_configuration.ts#L160

Reply:
I removed it from requiredPlugins.